dashboard/src/shared/schema.ts

```diff
@@ -1,17 +1,110 @@
+// WARN: yaml doesn't have updated definitions for TypeScript
+import * as YAML from "yaml";
 import { IBasicFormParam } from "./types";
 
-export function retrieveBasicFormParams(schema: any) {
-  // TBD
-  return {
-    username: {
-      name: "username",
-      path: "wordpressUsername",
-      value: "user",
-    } as IBasicFormParam,
-  };
+// Avoid to explicitly add "null" when an element is not defined
+import { nullOptions } from "yaml/types";
+nullOptions.nullStr = "";
+
+// retrieveBasicFormParams iterates over a JSON Schema properties looking for `form` keys
+// It uses the raw yaml to setup default values.
+// It returns a key:value map for easier handling.
+export function retrieveBasicFormParams(
+  defaultValues: string,
+  schema: any,
+): { [key: string]: IBasicFormParam } {
+  const basicFormParameters = {};
+  if (schema && schema.properties) {
+    const params = lookForFormParams(defaultValues, schema.properties);
+    if (params) {
+      params.forEach(p => (basicFormParameters[p.name] = p));
+    }
+  }
+  return basicFormParameters;
+}
+
+// getDocumentElem returns the YAML element given a root element
```

Comment:
Unless I'm missing something, `rootElem` here is just an object right? In which case, we can possibly remove (and not have to maintain) this function completely in favor of https://lodash.com/docs/#get ?

Reply:
Yeah, unfortunately it's not. It's a `YAML.ast.Document` from https://eemeli.org/yaml/#documents so we cannot use lodash methods.

Reply:
oh, nice catch, I understood that differently (I thought that the `*In` methods only worked with collections/arrays) but it's true that we can use that. Thanks!

The only caveat I have found is that `setIn` returns an error if the object doesn't define the parent element:

```
> d.toString()
'a: b\n'
> d.setIn(["b", "c"], "foo")
Error: Expected YAML collection at b. Remaining path: c
```

In any case, that's the same limitation that we have now.